src/ConfigParser.cc

```diff
@@ -361,6 +345,12 @@ ConfigParser::NextToken()
                 return nullptr;
             }
 
+            if (!path) {
+                debugs(3, DBG_CRITICAL, "FATAL: filename missing: " << token);
+                self_destruct();
+                return nullptr;
+            }
```

Comment:
I am repeating a resolved-without-comments-or-changes request from https://github.com/squid-cache/squid/pull/928#discussion_r797712934:

```suggestion
            assert(path); // a QuotedToken cannot be nil
```